From 15484dc9c24be214059b5bfecccb6f26bd79833c Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.i.king@gmail.com>
Date: Mon, 4 Nov 2024 21:51:19 +0000
Subject: [PATCH 5/8] core-shim: use shim'd types for shim_poll args

Signed-off-by: Colin Ian King <colin.i.king@gmail.com>
---
 core-shim.c | 4 ++--
 core-shim.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/core-shim.c b/core-shim.c
index 8f51991137e3..a7813474ab86 100644
--- a/core-shim.c
+++ b/core-shim.c
@@ -2831,8 +2831,8 @@ int shim_mseal(void *addr, size_t len, unsigned long flags)
  *	shim wrapper for ppoll
  */
 int shim_ppoll(
-	struct pollfd *fds,
-	nfds_t nfds,
+	shim_pollfd_t *fds,
+	shim_nfds_t nfds,
 	const struct timespec *tmo_p,
 	const sigset_t *sigmask)
 {
diff --git a/core-shim.h b/core-shim.h
index 2ec591f00ba4..1ea0910200c1 100644
--- a/core-shim.h
+++ b/core-shim.h
@@ -520,7 +520,7 @@ extern int shim_lstat(const char *pathname, struct stat *statbuf);
 extern int shim_stat(const char *pathname, struct stat *statbuf);
 extern unsigned char shim_dirent_type(const char *path, const struct dirent *d);
 extern int shim_mseal(void *addr, size_t len, unsigned long flags);
-extern int shim_ppoll(struct pollfd *fds, nfds_t nfds,
+extern int shim_ppoll(shim_pollfd_t *fds, shim_nfds_t nfds,
 	const struct timespec *tmo_p, const sigset_t *sigmask);
 
 #endif
-- 
2.47.0

