From 6df5cc75c900b6549c342927ff41a56f816b4169 Mon Sep 17 00:00:00 2001
From: John Kacur <jkacur@redhat.com>
Date: Fri, 15 Aug 2025 13:10:10 -0400
Subject: [PATCH] stress-ng: Add fPIC to Makefile

Add -fPIC to Makefile

Signed-off-by: John Kacur <jkacur@redhat.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6a1128ace84f..6fb61975be88 100644
--- a/Makefile
+++ b/Makefile
@@ -134,7 +134,7 @@ endif
 MACHINE := $(shell make -f Makefile.machine)
 ifneq ($(PRESERVE_CFLAGS),1)
 ifneq ($(MACHINE),$(filter $(MACHINE),alpha hppa ia64))
-flag = -Wformat -fstack-protector-strong -Werror=format-security
+flag = -Wformat -fstack-protector-strong -Werror=format-security -fPIC
 #
 # add -D_FORTIFY_SOURCE=2 if _FORTIFY_SOURCE is not already defined
 #
-- 
2.49.0

